x/fswap/types/msgs.go

```diff
@@ -75,5 +76,38 @@ func (m *MsgSwapAll) GetSigners() []sdk.AccAddress {
 
 // GetSignBytes implements the LegacyMsg.GetSignBytes method.
 func (m *MsgSwapAll) GetSignBytes() []byte {
-	return sdk.MustSortJSON(codec.ModuleCdc.MustMarshalJSON(m))
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(m))
+}
+
+func (m *MsgSetSwap) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Authority); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("invalid authority address: %s", m.Authority)
+	}
+
+	if err := m.Swap.ValidateBasic(); err != nil {
+		return err
+	}
+	if err := m.ToDenomMetadata.Validate(); err != nil {
+		return err
+	}
+	if m.Swap.ToDenom != m.ToDenomMetadata.Base {
+		return sdkerrors.ErrInvalidRequest.Wrapf("denomination does not match %s != %s", m.Swap.ToDenom, m.ToDenomMetadata.Base)
+	}
+
+	return nil
+}
+
+func (m *MsgSetSwap) GetSigners() []sdk.AccAddress {
+	signer := sdk.MustAccAddressFromBech32(m.Authority)
+	return []sdk.AccAddress{signer}
+}
+
+// GetSignBytes implements the LegacyMsg.GetSignBytes method.
+func (m *MsgSetSwap) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(m))
+}
+
+func (m *MsgSetSwap) String() string {
```

Comment:
It seems unnecessary.

Reply:
Should we implement this interface `sdk.Msg`??
I added for compiler's complain

Reply:
You should use the default stringer. Please remove the relevant option in the message.